### PR TITLE
add ability to open files directly without sharing closes #12

### LINKED ISSUE #12: [Feature Request] associate file types with the app
Hi!

**Use Case**
I download a document which gets saved on the device (there is no way for sharing without downloading in the app). On Android a notification will appear when the download is completed. I press the notification and the document opens in the default app for that filetype or if there is none, ask the user to select an app which is able to open the document type. The document gets opened and I click the buttons to share the document with paperless share. 

**Improvement**
If the paperless share app would be associated with for example application/pdf I could upload the document by just clicking the notification and choosing to open the file with paperless share. This would reduce the needed clicks significantly. Same when browsing files in an explorer. 

**Details**
https://pub.dev/packages/flutter_handle_file should make it possible to implement this with minimal effort for both iOS and Android.
If only Android is relevant it should work by just adding some intent-filter properties in the AndroidManifest.xml. 
I guess some code to handle the file path is also needed. 

Would you be interested in this feature? If yes, are you interested in a PR for this or do you want to have a look at it yourself?

### PR DESCRIPTION
**Pull request by NiFNi**
Hey @qcasey,
I've done something for #12. Feel free to have a look at it. I'll test it a bit more but the simple cases are working for me.
As you can see I added the intent-filter to allow any file. If the file is not supported the backend will return an error code which results in a Toast with the returned error message (for example that the content type application/zip is not supported). For me this seems better than adding all the supported file-types one by one as paperless supports a lot. If you would prefer to only allow opening of some specific files I can add that aswell.
I tried to use the plugin I linked in the issue but it seems to be pretty unmaintained. The last release is nearly 2 years ago and it was throwing some warnings. That's why I extracted the parts that were needed here. 
I am not sure if iOS should be handled somehow. If so I have no apple device so someone else would have to have a look at that. 

While at it I fixed some warnings and bumped some versions. The commits should be pretty easy to follow.

Edit: by bumping the encrypted_shared_preferences I think a new login is required when starting the app after the update.

### CODE REVIEW

**Comment by qcasey** [comment]
This looks great!

I'll test, merge, and release tomorrow if everything checks out. Thank you for your contribution

**Comment by NiFNi** [comment]
I found a case that's not working already.. After the author of the plugin merged my PR (https://github.com/nbonamy/flutter_handle_file/pull/1) I could get it working with that. There is still a warning. But the maintainer seems to be active and accepts PRs so I'll work on that next. But for now it seems to work :+1:

**Comment by NiFNi** [comment]
Alright, I fixed some more things in flutter_handle_file. I used my repo as dependency for now. I think that's fine until a new release of flutter_handle_file exists.

**Comment by qcasey** [comment]
Oops, committed to your main accidentally :laughing: sorry

I like how you handled the glob intent-filter in the app. However I think it's silly when Paperless Share pretends to handle some files like apks, .db, .mp4, etc..

I added specific mime types for the basic files paperless handles. The toast-on-error feature you wrote is still important, as the server will error on .docx and the like if Tika is not configured. 

What do you think?

**Comment by qcasey** [comment]
Other than those two notes this looks and tests great. I really appreciate the code cleanups and dep bumps as well.

**Comment by NiFNi** [comment]
All good! I think the intended way is that you can change stuff on the PR 😄
Yeah handling the files specifically is also fine for me. I was just to lazy to add all the files. So yeah looks good from my side aswell.

**Review by qcasey — APPROVED** [review]

File: lib/share.dart
Comment (NiFNi) [reply lib/share.dart]: Oh sorry.. I missed these comments. I can have a closer look at this. But currently it's just the behaviour the lib had before I did the changes. Maybe it's just an artefact of the lib where handle_file was based of. I think that was something with URIs and links.

**Pull request merged**
